torch/library.py

```diff
@@ -0,0 +1,74 @@
+import torch
+from ._ops import OpOverload
+from typing import Set
+import warnings
+
+__all__ = ['extend_library']
+
+# User created fragment libraries to extend existing libraries
+fragments_for_existing_libraries = {}
+
+# Set containing the combination of (namespace, operator, DispatchKey) for which a new kernel has been registered
+# The keys in the set are of the form `namespace + "/" + op_name + "/" + dispatch_key`.
+# This set is maintained to ensure that two libraries don't try to override the exact same functionality to avoid
+# libraries calling into kernels not intended to be called.
+impls: Set[str] = set()
+
+class Library:
+    # kind can be DEF, FRAGMENT (in C++ IMPL)
+    def __init__(self, kind, ns, dispatch_key=""):
+        # also prevent users from creating libraries with existing C++ libraries, e.g., aten
+        # if ns in existing_libraries and kind == "DEF":
+        #     raise ValueError("A library with name '" + ns + "' already exists. "
+        #                      "It's not allowed to have more than one library with the same name")
+        # elif kind == "FRAGMENT":
+        #     self.m = torch._C._dispatch_library(kind, ns, dispatch_key)
+        #     self.ns = ns
+        #     fragments_for_existing_libraries[id(self)] = self
+        #     self.op_impls = set()
+        # else:
+        #     raise ValueError("Unsupported kind: ", kind)
+        self.m = torch._C._dispatch_library(kind, ns, dispatch_key)
+        self.ns = ns
+        fragments_for_existing_libraries[id(self)] = self
+        self.op_impls = set()
+
+    def impl(self, op_name, dispatch_key, fn):
+        if isinstance(op_name, str):
+            name = self.ns + '.' + op_name
```

Comment:
If the ns was put on the C++ object it doesn't seem like this concatenation should be necessary